modules/backend/traits/PreferenceMaker.php

```diff
@@ -0,0 +1,136 @@
+<?php namespace Backend\Traits;
+
+use Str;
+use Backend\Models\UserPreference;
+
+/**
+ * Preference Maker Trait
+ *
+ * Adds methods for modifying user preferences in a controller class, or a class
+ * that contains a `$controller` property referencing a controller.
+ */
+trait PreferenceMaker
+{
+    /**
+     * Cache for retrieved user preferences.
+     *
+     * @var array
+     */
+    protected static $preferenceCache = [];
+
+    /**
+     * Saves a widget related key/value pair in to the users preferences
+     * @param string $key Unique key for the data store.
+     * @param mixed $value The value to store.
+     * @return void
+     */
+    public function putUserPreference(string $key, $value): void
+    {
+        $preferences = $this->getUserPreferences();
+        $preferences[$key] = $value;
+
+        $this->getStorage()::forUser()->set($this->getPreferenceKey(), $preferences);
```

Comment:
If the method is meant to be overrideable, should it be getPreferenceStorage()? Also, could it return an object instead of a class string?

Reply:
Good points - it's now changed.